compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -376,12 +378,11 @@ private BLangConstantValue calculateBitWiseOp(BLangConstantValue lhs, BLangConst
     }
 
     private BLangConstantValue calculateAddition(BLangConstantValue lhs, BLangConstantValue rhs) {
-        // TODO : Handle overflow in numeric values.
         Object result = null;
         switch (Types.getReferredType(this.currentConstSymbol.type).tag) {
             case TypeTags.INT:
             case TypeTags.BYTE: // Byte will be a compiler error.
-                result = (Long) lhs.value + (Long) rhs.value;
+                result = Math.addExact((Long) lhs.value, (Long) rhs.value);
```

Comment:
Here, we have to handle `byte` case separately. Please check the other operations also.

Reply:
Since byte scenario is not yet allowed, we can go with this change. Byte scenario will be fixed with https://github.com/ballerina-platform/ballerina-lang/issues/36538.